PWGHF/D2H/Tasks/taskDirectedFlowCharmHadrons.cxx

```diff
@@ -0,0 +1,304 @@
+  /// \param candidates are the selected candidates
+  template <DecayChannel channel, typename T1, typename Trk>
+  void runFlowAnalysis(CollsWithQvecs::iterator const& collision,
+                       T1 const& candidates, Trk const& /*tracks*/)
+  {
+    double cent = getCentrality(collision);
+    if (cent < centralityMin || cent > centralityMax) {
+      return;
+    }
+
+    if (!collision.triggerevent()) { // for selecting only callibrated events
+      return;
+    }
+
+    std::vector<double> qVecs = getQvec(collision);
+    double qxAQVec = qVecs[0];
+    double qyAQVec = qVecs[1];
+    double qxCQVec = qVecs[2];
+    double qyCQVec = qVecs[3];
+
+    auto qxZDCA = qxAQVec;
+    auto qyZDCA = qyAQVec;
+    auto qxZDCC = qxCQVec;
+    auto qyZDCC = qyCQVec;
+
+    auto QxtQxp = qxZDCC * qxZDCA;
+    auto QytQyp = qyZDCC * qyZDCA;
+    auto QxpQyt = qxZDCA * qyZDCC;
+    auto QxtQyp = qxZDCC * qyZDCA;
+
+    // denominators for SP calculation
+    registry.fill(HIST("hpQxtQxpvscent"), cent, QxtQxp);
+    registry.fill(HIST("hpQytQypvscent"), cent, QytQyp);
+    registry.fill(HIST("hpQxpQytvscent"), cent, QxpQyt);
+    registry.fill(HIST("hpQxtQypvscent"), cent, QxtQyp);
+
+    for (const auto& candidate : candidates) {
+      double massCand = 0.;
+      std::vector<double> outputMl = {-999., -999.};
+      if constexpr (std::is_same_v<T1, CandDplusData> || std::is_same_v<T1, CandDplusDataWMl>) {
+        massCand = hfHelper.invMassDplusToPiKPi(candidate);
+        if constexpr (std::is_same_v<T1, CandDplusDataWMl>) {
+          for (unsigned int iclass = 0; iclass < classMl->size(); iclass++)
+            outputMl[iclass] = candidate.mlProbDplusToPiKPi()[classMl->at(iclass)];
+        }
+      }
+
+      auto trackprong0 = candidate.template prong0_as<Trk>();
+      double sign = trackprong0.sign();
+
+      double ptCand = candidate.pt();
+      double etaCand = candidate.eta();
+      double phiCand = candidate.phi();
+      double cosNPhi = std::cos(phiCand);
+      double sinNPhi = std::sin(phiCand);
+
+      auto ux = cosNPhi;
+      auto uy = sinNPhi;
+      auto uxQxp = ux * qxZDCA;
+      auto uyQyp = uy * qyZDCA;
+      auto uxQxt = ux * qxZDCC;
+      auto uyQyt = uy * qyZDCC;
+
+      fillThn(massCand, ptCand, etaCand, cent, uxQxp, uyQyp, uxQxt, uyQyt, outputMl, sign);
+    }
+  }
+  // Dplus with ML
+  void processDplusMl(CollsWithQvecs::iterator const& collision,
+                      CandDplusDataWMl const& candidatesDplus, TracksWithExtra const& tracks)
+  {
+    runFlowAnalysis<DecayChannel::DplusToPiKPi>(collision, candidatesDplus, tracks);
+  }
+  PROCESS_SWITCH(HfTaskDirectedFlowCharmHadrons, processDplusMl, "Process Dplus candidates with ML", false);
+
+  // Dplus with rectangular cuts
+  void processDplus(CollsWithQvecs::iterator const& collision,
+                    CandDplusData const& candidatesDplus, TracksWithExtra const& tracks)
```

Comment:
Put one argument per line.

Reply:
[Reply]: Added few arguments.

Reply:
I don't see a change.

Reply:
[Reply]: I have added few comments in the code such as "// correlations in the denominators for SP calculation" in Line 160 and so on....What specific comments/arguments you want in each line of code?

Reply:
I didn't mention comments. I meant the arguments of the process function. It's easier to read the table subscriptions when they are broken into one argument per line.

Reply:
Is it okay now?